jolt-core/src/r1cs/jolt_constraints.rs

```diff
@@ -150,53 +160,376 @@ impl<F: JoltField> R1CSConstraintBuilder<F> for JoltConstraints {
+        //           (memory_write[2] - lookup_output) * remainder * (remainder - 1) * (remainder - 3) +
+        //           (memory_write[3] - lookup_output) * remainder * (remainder - 1) * (remainder - 2)
+        //       ] = 0
+        let write0_minus_lookupoutput = cs.allocate_prod(
+            LC::sub2(JoltIn::RAM_Write_Byte0, JoltIn::LookupOutput),
+            remainder123,
+        );
+        let write1_minus_lookupoutput = cs.allocate_prod(
+            LC::sub2(JoltIn::RAM_Write_Byte1, JoltIn::LookupOutput),
+            remainder023,
+        );
+        let write2_minus_lookupoutput = cs.allocate_prod(
+            LC::sub2(JoltIn::RAM_Write_Byte2, JoltIn::LookupOutput),
+            remainder013,
+        );
+        let write3_minus_lookupoutput = cs.allocate_prod(
+            LC::sub2(JoltIn::RAM_Write_Byte3, JoltIn::LookupOutput),
+            remainder012,
+        );
+        let term0 = LC::sum_any(vec![
+            write0_minus_lookupoutput,
+            write1_minus_lookupoutput,
+            write2_minus_lookupoutput,
+            write3_minus_lookupoutput,
+        ]);
+        cs.constrain_prod(JoltIn::OpFlags_IsSb, term0, LC::zero());
+
+        // STORE CONSTRAINT b-2
+        // (SH_flag) [
+        //           (memory_write[0] + 2^{8}memory_write[1] - lookup_output) * (remainder - 2)  +
+        //           (memory_write[2] +  2^{8}*memory_write[3] - lookup_output) * remainder
+        //     ] = 0
+        let write01_memory = LC::sum2(
+            cs.create_term(JoltIn::RAM_Write_Byte0, 1),
+            cs.create_term(JoltIn::RAM_Write_Byte1, 1 << 8),
+        );
+        let write23_memory = LC::sum2(
+            cs.create_term(JoltIn::RAM_Write_Byte2, 1),
+            cs.create_term(JoltIn::RAM_Write_Byte3, 1 << 8),
+        );
+        let write01_minus_lookupoutput = cs.allocate_prod(
+            LC::sub2(write01_memory.clone(), JoltIn::LookupOutput),
+            remainder_minus_2_term.clone(),
+        );
+        let write23_minus_lookupoutput = cs.allocate_prod(
+            LC::sub2(write23_memory.clone(), JoltIn::LookupOutput),
+            JoltIn::Remainder,
+        );
+        let term1 = LC::sum2(write01_minus_lookupoutput, write23_minus_lookupoutput);
+        cs.constrain_prod(JoltIn::OpFlags_IsSh, term1, LC::zero());
+
+        // STORE CONSTRAINT b-3
+        // (SW_flag) [
+        //           memory_write[0] + 2^{8}memory_write[1] +
+        //           2^{16}memory_write[2] + 2^{24}memory_write[3]  -
+        //           packed_query)
+        //           ] = 0
+
+        let write_memory = LC::sum_any(vec![
+            cs.create_term(JoltIn::RAM_Write_Byte0, 1),
+            cs.create_term(JoltIn::RAM_Write_Byte1, 1 << 8),
+            cs.create_term(JoltIn::RAM_Write_Byte2, 1 << 16),
+            cs.create_term(JoltIn::RAM_Write_Byte3, 1 << 24),
+        ]);
+        cs.constrain_prod(
```

Comment:
We should make a `cs.constrain_pack_be_conditional` so the format would look as follows. 
```
cs.constrain_pack_be_conditional(JoltIn::OpFlags_IsSw, input_range!(JoltIn::RAM_Write_Byte0, JoltIn::RAM_Write_Byte3), packed_query, 8)
```

There's a `TODO(sragss)` above to do this. I can add, leave this comment open.